--- a/net/minecraft/world/phys/shapes/VoxelShape.java
+++ b/net/minecraft/world/phys/shapes/VoxelShape.java
@@ -433,6 +_,12 @@
         );
     }
     // Paper end - optimise collisions
+    // Sakura start - protect block shapes against plugins
+    public final VoxelShape copy() {
+        this.cachedToAABBs = null;
+        return this.move(Vec3.ZERO);
+    }
+    // Sakura end - protect block shapes against plugins
 
     protected VoxelShape(DiscreteVoxelShape shape) {
         this.shape = shape;
