--- a/net/minecraft/world/level/block/LadderBlock.java
+++ b/net/minecraft/world/level/block/LadderBlock.java
@@ -29,6 +_,12 @@
     protected static final VoxelShape WEST_AABB = Block.box(13.0, 0.0, 0.0, 16.0, 16.0, 16.0);
     protected static final VoxelShape SOUTH_AABB = Block.box(0.0, 0.0, 0.0, 16.0, 16.0, 3.0);
     protected static final VoxelShape NORTH_AABB = Block.box(0.0, 0.0, 13.0, 16.0, 16.0, 16.0);
+    // Sakura start - protect block shapes against plugins
+    private static final VoxelShape EAST_AABB_COPY = EAST_AABB.copy();
+    private static final VoxelShape WEST_AABB_COPY = WEST_AABB.copy();
+    private static final VoxelShape SOUTH_AABB_COPY = SOUTH_AABB.copy();
+    private static final VoxelShape NORTH_AABB_COPY = NORTH_AABB.copy();
+    // Sakura end - protect block shapes against plugins
 
     @Override
     public MapCodec<LadderBlock> codec() {
@@ -44,14 +_,16 @@
     protected VoxelShape getShape(BlockState state, BlockGetter level, BlockPos pos, CollisionContext context) {
         switch ((Direction)state.getValue(FACING)) {
             case NORTH:
-                return NORTH_AABB;
+                // Sakura start - protect block shapes against plugins
+                return NORTH_AABB_COPY;
             case SOUTH:
-                return SOUTH_AABB;
+                return SOUTH_AABB_COPY;
             case WEST:
-                return WEST_AABB;
+                return WEST_AABB_COPY;
             case EAST:
             default:
-                return EAST_AABB;
+                return EAST_AABB_COPY;
+                // Sakura end - protect block shapes against plugins
         }
     }
 
