--- a/net/minecraft/network/protocol/game/ClientboundContainerSetContentPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundContainerSetContentPacket.java
@@ -23,10 +_,10 @@
         this.items = NonNullList.withSize(items.size(), ItemStack.EMPTY);
 
         for (int i = 0; i < items.size(); i++) {
-            this.items.set(i, items.get(i).copy());
+            this.items.set(i, items.get(i).copyForPacket()); // Sakura - modify components sent to the client
         }
 
-        this.carriedItem = carriedItem.copy();
+        this.carriedItem = carriedItem.copyForPacket(); // Sakura - modify components sent to the client
     }
 
     private ClientboundContainerSetContentPacket(RegistryFriendlyByteBuf buffer) {
