--- a/net/minecraft/world/level/redstone/ExperimentalRedstoneUtils.java
+++ b/net/minecraft/world/level/redstone/ExperimentalRedstoneUtils.java
@@ -18,6 +_,7 @@
                 orientation = orientation.withFront(front);
             }
             // Paper start - Optimize redstone (Alternate Current) - use default front instead of random
+            // Sakura - redstone implementation api; conflict on change
             else if (level.paperConfig().misc.redstoneImplementation == io.papermc.paper.configuration.WorldConfiguration.Misc.RedstoneImplementation.ALTERNATE_CURRENT) {
                 orientation = orientation.withFront(Direction.WEST);
             }
